src/AMSWorkflow/ams/stage.py

```
@@ -691,10 +741,23 @@ def _parallel_execute(self, exec_vehicle_cls):
         for a in self._tasks:
             self._executors.append(exec_vehicle_cls(target=a))
 
-        for e in self._executors:
+        pids_to_kill = []
+        for e, a in zip(self._executors, self._tasks):
+            print(f"Started executor {a}")
             e.start()
 
+        if isinstance(self._tasks[0], RMQDomainDataLoaderTask):
+            print("Found Head RMQ Loader")
+            pids_to_kill.append(self._executors[0].pid)
+        else:
+            print("Expected 0 to be RMQ, but was", self._tasks[0].__clas__.__name__)
```

Should we make this an assertion?

---

Is it a state that we cannot recover from? If yes, we should throw an error or put an assert

---

We should fail yes. Although this is me being very cautious with what I wrote. I will modify this in an assertion.